packages/compiler/server/serverlib.ts

```diff
@@ -670,8 +700,20 @@ export function createServer(host: ServerHost): Server {
     }
 
     const references: IdentifierNode[] = [];
-    for (const script of program.sourceFiles.values() ?? []) {
-      visitChildren(script, function visit(node) {
+    if (wholeProgram === true) {
+      for (const script of program.sourceFiles.values() ?? []) {
+        visitChildren(script, function visit(node) {
```

Comment:
I think you should be able to have some common code here instead of duplicating. Could even probably just callback to `findReferenceIdentifiers` with wholeProgram set to `false`

Reply:
sounds good